### PR TITLE
Move server binaries to /cmd directory

### PR DESCRIPTION
**Pull request by RJPercival**
Also moves /server/main.go to /cmd/server, because this part of the "server" package is fairly specific to those binaries.

### Checklist

- [x] I have updated the [CHANGELOG](CHANGELOG.md).
  - Adjust the draft version number according to [semantic versioning](https://semver.org/) rules.
- [x] I have updated [documentation](docs/) accordingly (including the [feature implementation matrix](docs/Feature_Implementation_Matrix.md)).

### CODE REVIEW

File: cmd/internal/serverutil/server.go [hunk 1]
```diff
@@ -12,8 +12,8 @@
 // See the License for the specific language governing permissions and
```
Comment (gdbelvin) [inline cmd/internal/serverutil/server.go]: Please name this file `main.go` or `serverutil.go`

File: examples/deployment/aws/terraform.tf [hunk 2]
```diff
@@ -132,7 +132,7 @@ mkdir -p /go
 export GOPATH=/go
 
 # Install Trillian
-go get github.com/google/trillian/server/trillian_log_server
+go get github.com/google/trillian/cmd/server/trillian_log_server
```
Comment (gdbelvin) [inline examples/deployment/aws/terraform.tf]:
```suggestion
go get github.com/google/trillian/cmd/trillian_log_server
```

File: examples/deployment/docker/log_server/Dockerfile [hunk 3]
```diff
@@ -14,9 +14,9 @@ RUN go mod download
 COPY . .
 
 # Build the server.
-RUN go get ./server/trillian_log_server
+RUN go get ./cmd/server/trillian_log_server
```
Comment (gdbelvin) [inline examples/deployment/docker/log_server/Dockerfile]:
```suggestion
RUN go get ./cmd/trillian_log_server
```

Comment (gdbelvin) [reply examples/deployment/docker/log_server/Dockerfile]: and throughout

File: cmd/internal/serverutil/server.go (hunk 1 above)
Comment (RJPercival) [reply cmd/internal/serverutil/server.go]: Renamed to main.go.

**Pull request merged**